### PR TITLE
Add failing test for rector/rector#7209

### PR DESCRIPTION
**Pull request by eigan**
https://github.com/rectorphp/rector/issues/7209

### CODE REVIEW

File: rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/skip_rename_class_different_namespace.php.inc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+{
+    /**
+     * @var Storage
+     */
+    protected $storage;
+
+    public function __construct(
+        Storage $storage
+    ) {
+        \Log::error('Some loggin');
```
Comment (samsonasik) [inline rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/skip_rename_class_different_namespace.php.inc:17]: import \Log is expected, except you disable it https://github.com/rectorphp/rector-src/blob/b1583c7d8636aa32e6d5f7bdf37162e3e38a3907/packages/Config/RectorConfig.php#L84

**Comment by eigan** [comment]
image

Not sure if related at all.

File: rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/skip_rename_class_different_namespace.php.inc (hunk 1 above)
Comment (samsonasik) [reply rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/skip_rename_class_different_namespace.php.inc:17]: You can update to before -> after fixture that make `\Log` still imported while keeping docblock like this

https://github.com/rectorphp/rector-src/blob/main/rules-tests/Renaming/Rector/Name/RenameClassRector/FixtureAutoImportNames/instanceof_import.php.inc

**Comment by eigan** [comment]
image

I believe the issue is somewhere here. The var tag for Storage seems to not resolve the proper namespace.

**Comment by eigan** [comment]
image

Current node while working with the `@var` is `Foo`, which makes it impossible to get the correct namespace when calling `resolveNamespacedName`. I believe it must be `Bar` or something to resolve the correct `uses`.


This is way out of my comfort zone and I have no idea what I am looking at. So stopping now, hope it helped anyway.

**Comment by samsonasik** [comment]
please update the test to allow import `\Log` as expected https://github.com/rectorphp/rector-src/pull/2440#discussion_r889744749

**Comment by samsonasik** [comment]
@eigan I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/2441

**Pull request closed without merging**